### DISCUSSION TITLE
'Truffle test' does not run（truffle@5.1.31）

### DISCUSSION THREAD

**Discussion by okahijiki** [ConsenSys-archive/truffle#5554] [General Q&A]
‘truffle test’ does not run at all.

I have written the code in truffle@5.1.31, according to the book I am tutorializing.

//「Hands-On Smart Contract Development With Solidity and Ethereum」(O'Reilly/Japanese edition )

The 'truffle test' can be viewed here.
[https://youtu.be/rbj-SdMOObY](https://youtu.be/rbj-SdMOObY)

The code is here.
https://github.com/okahijiki/fundraiser

Please let me know how to deal with this problem.

thank you,

**Comment by okahijiki** [comment]
Is there a relationship between truffle unbox react and node version?
The current node version is 16.15.1

**Reply by apdan510** [reply]
@okahijiki Node 16.15.1 should work. Do you get any error messages when you enter truffle test in your terminal?

**Reply by okahijiki** [reply]
@apdan510

Truffle test does not run the test(Truffle v5.1.31）

Next, I tried changing the Truffle version(Truffle v5.5.30）
The error message in that case is as follows.

CompileError: project:/contracts/Fundraiser.sol:9:1: ParserError: Function, variable, struct or modifier declaration expected.

If it was caused by something other than Truffle test, please excuse me.

turuffle test can be viewed on Youtube below.

[https://youtu.be/MifAGZnbSrk](https://youtu.be/MifAGZnbSrk)

thnk you,

**Reply by apdan510** [reply]
Can you please share your smart contract? I would love to take a look at it. 

Thank you!
-Alex

**Reply by okahijiki** [reply]
https://github.com/okahijiki/fundraiser

Thank you,
-@okahijiki

**Reply by apdan510** [reply]
@okahijiki looks like the fundraiser.sol contract has some errors. As a test, can you please remove this line and run truffle test again?

```
constructor(string memory _name)public{
        name = _name;
```

**Reply by okahijiki** [reply]
Thank you for your response.
I removed the code you point out.

In truffle v5.1.31, the test did not work at all, as well as without the deletion.

In Truffle v5.5.31, I get different error messages when I remove it and when I do not remove it.


・Deleted Cases

solc: 0.5.16+commit.9c3226ce.Emscripten.clang

/Users/tsuru/.nvm/versions/node/v16.15.1/lib/node_modules/truffle/build/3539.bundled.js:22177

throw new Error("Could not find artifacts for " + import_path + " from any sources");

Error: Could not find artifacts for SimpleStorage from any sources…….

・Cases not to delete

CompileError: project:/contracts/Fundraiser.sol:9:1: ParserError: Function, variable, struct or modifier declaration expected…….

For more information about Truffle test, please refer to the following YouTube
[https://youtu.be/8byupjw2FNM](https://youtu.be/8byupjw2FNM)

**Reply by apdan510** [reply]
Let me dig into this a bit

**Reply by okahijiki** [reply]
Thank you for your time.

I found one mistake in this discussion after I left my last comment.
The last { was missing on Fundraiser.sol, as shown below.

`
pragma solidity >0.4.23 <0.7.0;

contract Fundraiser{
    string public name;

    constructor(string memory _name)public{
        name = _name;
    }

`


I have now corrected the mistake and have the following.

`
pragma solidity >0.4.23 <0.7.0;

contract Fundraiser{
    string public name;

    constructor(string memory _name)public{
        name = _name;
    }
}

`

The code I have on GitHub has also been corrected.
[https://github.com/okahijiki/fundraiser/blob/main/contracts/Fundraiser.sol](url)

As a result, the error code has also changed as follows.

[Before correction]
CompileError: project:/contracts/Fundraiser.sol:9:1: ParserError: Function, variable, struct or modifier declaration expected.

[After correction]
throw new Error("Could not find artifacts for " + import_path + " from any sources");
Error: Could not find artifacts for SimpleStorage from any sources

I just wanted to inform you.

**Reply by okahijiki** [reply]
Deleted 「migrations/2.deploy.contracts.js」and the problem is now clear. I am sorry for the trouble.
